File: Src2/Icm.Core/Text/UsualRegularExpressions.vb
Namespace Icm.Text.RegularExpressions

    Public Module UsualRegularExpressions
        Public Const regexReal As String = "[+-]?\d+([\.,]\d*)?"
        Public Const regexInteger As String = "[+-]?\d+"
        Public Const regexOctal As String = "[+-]?[0-7]+"
        Public Const regexHexadecimal As String = "[+-]?[0-9a-fA-F]+"

        Public Const DateOnlyValidator As String = "\d?\d/\d?\d/([1-9]\d)?\d\d"
        Public Const DateAndTimeValidator As String = "\d?\d/\d?\d/(\d\d)?\d\d\s+\d?\d:\d\d"
        Public Const TimeOnlyValidator As String = "\d?\d:\d\d"
        Public Const DateOrTimeValidator As String = "\d?\d/\d?\d/(\d\d)?\d\d(\s+\d?\d:\d\d)?"

        Public Const RealValidator As String = "^[+-]?\d+([.,]\d*)?$"
        Public Const IntegerValidator As String = "\d+"

    End Module


End Namespace

File: Src/Icm.Core/Text/UsualRegularExpressions.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
namespace Icm.Text.RegularExpressions
{

	public static class UsualRegularExpressions
	{
		public const string regexReal = "[+-]?\\d+([\\.,]\\d*)?";
		public const string regexInteger = "[+-]?\\d+";
		public const string regexOctal = "[+-]?[0-7]+";

		public const string regexHexadecimal = "[+-]?[0-9a-fA-F]+";
		public const string DateOnlyValidator = "\\d?\\d/\\d?\\d/([1-9]\\d)?\\d\\d";
		public const string DateAndTimeValidator = "\\d?\\d/\\d?\\d/(\\d\\d)?\\d\\d\\s+\\d?\\d:\\d\\d";
		public const string TimeOnlyValidator = "\\d?\\d:\\d\\d";

		public const string DateOrTimeValidator = "\\d?\\d/\\d?\\d/(\\d\\d)?\\d\\d(\\s+\\d?\\d:\\d\\d)?";
		public const string RealValidator = "^[+-]?\\d+([.,]\\d*)?$";

		public const string IntegerValidator = "\\d+";
	}


}


//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
